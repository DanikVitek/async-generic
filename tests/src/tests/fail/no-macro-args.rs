use async_generic::async_generic;

#[async_generic(Send)]
fn do_stuff() -> String {
    if _async {
        my_async_stuff().await
    } else {
        "not async".to_owned()
    }
}

async fn my_async_stuff() -> String {
    "async".to_owned()
}

#[async_std::main]
async fn main() {}
