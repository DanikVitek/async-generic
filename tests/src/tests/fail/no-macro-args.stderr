error: async_generic can only take an async_signature argument
 --> src/tests/fail/no-macro-args.rs:3:17
  |
3 | #[async_generic(Send)]
  |                 ^^^^
